src/control/tools/StrokeStabilizer/ArithmeticStrokeStabilizer.cpp

```diff
@@ -0,0 +1,70 @@
+#include "ArithmeticStrokeStabilizer.h"
+
+#include <cmath>
+
+// #include <control/XournalMain.h>
+// #include <gdk/gdk.h>
+
+#include "control/tools/StrokeHandler.h"  // TODO Ugly fix, remove!
+
+// Needed?
+// #include "control/settings/Settings.h"
+// #include "config-features.h"
+
+ArithmeticStrokeStabilizer::ArithmeticStrokeStabilizer(const PositionInputData& pos):
+        // TODO read config
+        bufferLength(StrokeHandler::stabilizingBufferSize) {
+    /**
+     * Fill the queue with copies of our starting point
+     * Is this the best way to do this? We could also aggregate the first events as they come
+     * but this would add a test (on the length of the queue) at every event triggered.
+     */
+    for (int i = 1; i < bufferLength; i++) {
+        // The next event will be pushed before averaging, so there will be exactly bufferLength elements
+        // in the queue then.
+        eventBuffer.emplace_front(pos.x, pos.y, pos.pressure);
+    }
+    g_message("Created ArithmeticStrokeStabilizer with bufferLength = %d", bufferLength);
+}
+
+void ArithmeticStrokeStabilizer::stabilizePointUsingEvent(const PositionInputData& pos, double zoom, Point* point) {
+
+    /**
+     * Create and push the event
+     */
+    eventBuffer.emplace_front(pos.x, pos.y, pos.pressure);
+
+    /**
+     * Average the coordinates using an arithmetic mean
+     */
+    double sumOfX = 0;
+    double sumOfY = 0;
+    double sumOfPressures = 0;
+
+    // Type of it: std::deque<BufferedEvent>::const_iterator
+    for (auto it = eventBuffer.cbegin(); it != eventBuffer.cend(); ++it) {
+        sumOfX += (*it).x;
+        sumOfY += (*it).y;
+        sumOfPressures += (*it).pressure;
+    }
+
+    /**
+     * Pop the oldest event to keep the bufferLength constant
+     */
+    eventBuffer.pop_back();
+
+    /**
+     * Rescale the averaged coordinates before assigning them to the point
+     */
+    point->x = sumOfX / ((double)bufferLength * zoom);
+    point->y = sumOfY / ((double)bufferLength * zoom);
+    point->z = sumOfPressures / (double)bufferLength;
+}
+
+void ArithmeticStrokeStabilizer::pushEvent(const PositionInputData& pos) {
+    eventBuffer.emplace_front(pos.x, pos.y, pos.pressure);
+    /**
+     * Keep the length of the buffer constant
+     */
+    eventBuffer.pop_back();
```

Comment:
First pop then emplace. 
In some corner cases, your version could add a bucket to the deque.
Also, it could remove the currently added item.

Reply:
Thanks for the advice! The arithmetic stabilizer will most likely not make it to the final code (I really don't see the point of keeping it). For now, it's just for testing.
Also, in that specific case, the data should not be stored in a deque in the first place. It should be a circular buffer which are much faster (and don't pop/push but simply overwrite).

Reply:
Using a deque for prototyping is imo the best decision someone can make. It's nearly as fast as all vector operations and way faster on front insertion (same as back insertion). Also, it supports nearly all Container OP's. So go ahead and finish your prototyping. The queues can then replaced later by something like a [boost::circular_buffer](https://www.boost.org/doc/libs/1_61_0/doc/html/circular_buffer.html). Don't hang up on the internal datatype :).